torchtitan/models/llama/model.py

```diff
@@ -425,12 +425,14 @@ def forward(self, tokens: torch.Tensor):
             torch.Tensor: Output logits after applying the Transformer model.
 
         """
-        h = self.tok_embeddings(tokens)
+        # passthrough for nonexistent layers, allows easy configuration of pipeline parallel stages
+        h = self.tok_embeddings(tokens) if self.tok_embeddings else tokens
 
-        for layer in self.layers:
+        for layer in self.layers.values():
```

Comment:
is order still respected after switching to dict?

Reply:
yea, it is. https://pytorch.org/docs/stable/generated/torch.nn.ModuleDict.html